tests/Affects/Mails/Unit/ConfigureMailsTest.php

```diff
@@ -16,13 +16,49 @@
 
 namespace Tenancy\Tests\Affects\Mails\Unit;
 
+use Illuminate\Mail\Transport\ArrayTransport;
+use Mockery;
 use Tenancy\Affects\Mails\Events\ConfigureMails;
 use Tenancy\Affects\Mails\Provider;
+use Tenancy\Facades\Tenancy;
 use Tenancy\Tests\Affects\AffectsEventUnitTestCase;
 
 class ConfigureMailsTest extends AffectsEventUnitTestCase
 {
     protected $affectsProvider = Provider::class;
 
     protected $event = ConfigureMails::class;
+
+    /** @test */
+    public function the_event_can_replace_swiftmailer()
+    {
+        $this->app->register($this->affectsProvider);
+
+        $this->events->listen($this->event, function (ConfigureMails $event) {
+            $event->replaceSwiftMailer(new ArrayTransport());
+
+            $this->assertInstanceOf(ArrayTransport::class, $event->mailer->getSwiftMailer()->getTransport());
+        });
+
+        Tenancy::setTenant($this->tenant);
+    }
+
+    /** @test */
+    public function the_event_can_delegate_to_mailer()
+    {
+
+        $this->app->register($this->affectsProvider);
+
+        $this->events->listen($this->event, function (ConfigureMails $e) {
+            $mailerSpy = Mockery::spy(get_class($e->mailer));
+
+            $e->mailer = $mailerSpy;
+
+            $e->to('test@example.com');
+
+            $mailerSpy->shouldHaveReceived()->to('test@example.com');
+        });
```

Comment:
Same here, it feels really weird to me to put all the testing code in the listener of the affects. I feel like tests should run outside of the listeners when possible.